test/e2e/Tests/Tests/VersioningTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class VersioningTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+
+    public VersioningTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+    }
+
+    [Theory]
+    [InlineData("")] // Represents a non-versioned case.
+    [InlineData("1.0")]
+    [InlineData("2.0")]
+    public async Task TestVersionedOrchestration_OKWithMatchingVersion(string version)
```

Comment:
Please consider adding a test that would check which version is passed to suborchestrations. In fact, there seems to be a problem here: in my manual tests, `context.Version` is empty in suborchestrations, but I expected it to match `defaultVersion` in `host.json`. See the discussion here: https://github.com/Azure/azure-functions-durable-extension/pull/3072#discussion_r2034075880

Reply:
Added tests (and the required code) for this use case.